soroban-env-host/src/host/error.rs

```diff
@@ -229,40 +214,48 @@ impl Host {
     /// enriches the returned [Error] with [DebugInfo] in the form of a
     /// [Backtrace] and snapshot of the [Events] buffer.
     pub fn error(&self, error: Error, msg: &str, args: &[Val]) -> HostError {
-        if let Ok(true) = self.is_debug() {
-            // We _try_ to take a mutable borrow of the events buffer refcell
-            // while building up the event we're going to emit into the events
-            // log, failing gracefully (just emitting a no-debug-info
-            // `HostError` wrapping the supplied `Error`) if we cannot acquire
-            // the refcell. This is to handle the "double fault" case where we
-            // get an error _while performing_ any of the steps needed to record
-            // an error as an event, below.
-            if let Ok(mut events_refmut) = self.0.events.try_borrow_mut() {
-                if let Err(e) = self.err_diagnostics(events_refmut.deref_mut(), error, msg, args) {
-                    return e;
+        self.with_debug_budget(
+            || {
+                // We _try_ to take a mutable borrow of the events buffer refcell
+                // while building up the event we're going to emit into the events
+                // log, failing gracefully (just emitting a no-debug-info
+                // `HostError` wrapping the supplied `Error`) if we cannot acquire
+                // the refcell. This is to handle the "double fault" case where we
+                // get an error _while performing_ any of the steps needed to record
+                // an error as an event, below.
+                // We do the event-borrowing ourselves here rather than calling
```

Comment:
This comment got extended with a copy-paste of a comment from elsewhere that's restating what the comment above also says (that we're doing a fallible borrow) but incorrectly since we actually _don't_ want to assume we're in a shared-borrow here, we're assuming we're _not_ in a _mutable_ borrow, but are guarding against it happening by accident.

Reply:
Removed it (as well as the one above).